### PR TITLE
ICU-20568 Read unitsTest.txt, prepare to run unit tests.

### PR DESCRIPTION
**Pull request by hugovdm**
I'm using "WIP" to flag "Work In Progress". There are no "WIP"s in the codebase yet. :-)

This Pull Request disables unit tests whose identifiers aren't supported yet: to be re-enabled once MeasureUnit is updated.

##### Checklist

- [X] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20568
- [X] Updated PR title and link in previous line to include Issue number
- [X] Issue accepted
- [X] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/test/intltest/unitstest.cpp [hunk 1]
```diff
@@ -160,4 +166,102 @@ void UnitsTest::testArea() {
...
+    return;
+  }
+
+  CharString path(sourceTestDataPath, errorCode);
+  path.appendPathPart("units", errorCode);
+  path.appendPathPart("unitsTest.txt", errorCode);
+
+  u_parseDelimitedFile(path.data(), ';', fields, kNumFields,
+                       unitsTestDataLineFn, this, errorCode);
+  if (U_FAILURE(errorCode)) {
```
Comment (sffc) [inline icu4c/source/test/intltest/unitstest.cpp]: errorCode is an IcuTestErrorCode.  You can use `errIfFailureAndReset`

File: icu4c/source/test/intltest/unitstest.cpp [hunk 2]
```diff
@@ -160,4 +166,102 @@ void UnitsTest::testArea() {
...
+  StringPiece y = trimField(fields[2]);
+  StringPiece commentConversionFormula = trimField(fields[3]);
+  StringPiece utf8Expected = trimField(fields[4]);
+
+  UNumberFormat *nf =
+      unum_open(UNUM_DEFAULT, NULL, -1, "en_US", NULL, pErrorCode);
+  UnicodeString uExpected = UnicodeString::fromUTF8(utf8Expected);
+  double expected = unum_parseDouble(nf, uExpected.getBuffer(),
+                                     uExpected.length(), 0, pErrorCode);
+  unum_close(nf);
```
Comment (sffc) [inline icu4c/source/test/intltest/unitstest.cpp:206]: Technically this works, but please use the following to convert a string to a double…

```cpp
DecimalQuantity expected;
expected.setToDecNumber({uExpected.getBuffer(), uExpected.length()}, *pErrorCode);
double expectedDouble = expected.toDouble();
```

**Review by sffc — CHANGES_REQUESTED** [review]
Good milestone; just a couple things

File: icu4c/source/test/intltest/unitstest.cpp (hunk 2 above)
Comment (hugovdm) [reply icu4c/source/test/intltest/unitstest.cpp:206]: I'm currently still parsing "1,000.00". When I do this with DecimalQuantity, it gives U_DECIMAL_NUMBER_SYNTAX_ERROR. Is it better to use DecimalQuantity and manually strip out the thousands separators? It seemed reasonable to me to simply treat "1,000.00" as a localised number format.

Comment (sffc) [reply icu4c/source/test/intltest/unitstest.cpp:206]: OK.  Fine to leave it as-is.

**Review by sffc — APPROVED** [review]

**Pull request merged**